File: 89. Gray Code.cpp
#include <cmath>
class Solution {
public:
    vector<int> grayCode(int n) {
        vector <int> res (1);
        for (int i = 0; i < n; i++)
        {
            vector <int> temp;
            for (int j = 0; j < res.size(); j++)
                temp.push_back(res[j]);
            for (int j = res.size() - 1; j >= 0; j--)
                temp.push_back(res[j] + pow(2, i));
            res = temp;
        }
        return res;
    }
};

File: 89. Gray Code.py
class Solution(object):
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        res = [0]
        for i in xrange(n):
            temp = []
            for j in xrange(len(res)):
                temp += [res[j]]
            for j in xrange(len(res)-1, -1, -1):
                temp += [res[j] + 2**i]
            res[:] = temp[:]
        return res
